### PR TITLE
pinns: fix ownership for created namespaces

### PR DESCRIPTION
**Pull request by giuseppe**
when creating a user namespace, make sure the user namespace is
created first and the right mappings are assigned.  Once that is done,
create the remaining namespaces so that they are owned by root in the
user namespace.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

#### What type of PR is this?

> /kind api-change
/kind bug
> /kind cleanup
> /kind dependency-change
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

#### What this PR does / why we need it:

we can use user namespaces and selinux with managed ns lifecycle

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pinns/src/pinns.c [hunk 1]
```diff
@@ -142,20 +142,39 @@ int main(int argc, char **argv) {
     }
   } else {
     /* if we create a user namespace, we need a new process.  */
-    if (pipe2(p, O_DIRECT) < 0)
-      pexit("pipe");
+    if (socketpair(AF_UNIX, SOCK_SEQPACKET | SOCK_CLOEXEC, 0, p))
```
Comment (mrunalp) [inline pinns/src/pinns.c:145]: Can you split this change into a separate commit?

Comment (giuseppe) [reply pinns/src/pinns.c:145]: sure.  Pushed now

**Comment by kolyshkin** [comment]
CI failures are all the same; there's an issue about it: https://github.com/kubernetes/kubernetes/issues/96222

**Review by saschagrunert — APPROVED** [review]

**Comment by kolyshkin** [comment]
> CI failures are all the same; there's an issue about it: kubernetes/kubernetes#96222

The issue is fixed, so

/retest

**Comment by kolyshkin** [comment]
integration failure is unrelated; restarted pipeline ci

not ok 167 stats
 (in test file ./stats.bats, line 17)
   `id=$(crictl run "$TESTDATA"/container_redis.json "$TESTDATA"/sandbox_config.json)' failed
 time="2020-11-05T09:39:30Z" level=info msg="Starting CRI-O, version: 1.20.0-dev, git: 3ca6f8daeec3f1667e99b61298819ecc7b31c77f(dirty)"
 time="2020-11-05T09:39:46Z" level=fatal msg="running container: creating container failed: rpc error: code = Unknown desc = Error reading blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4: Get \"https://quay.io/v2/crio/redis/blobs/sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\": EOF"

**Review by kolyshkin — APPROVED** [review]
/lgtm
